cl/assets/react/DisclosureList.tsx

```diff
@@ -0,0 +1,227 @@
+import React from 'react';
+import { appFetch } from './_fetch';
+import { UserState } from './_types';
+import debounce from 'lodash.debounce';
+
+interface Row {
+  id: number;
+  latest_disclosure_url: string;
+  name_full: string;
+  position_str: string;
+  disclosure_years: string;
+  thumbnail_path: string;
+}
+
+const DisclosureList: React.FC<UserState> = () => {
+  const [data, setData] = React.useState<Row[]>([]);
+  const [query, setQuery] = React.useState('');
+
+  React.useEffect(() => {
+    const handleEsc = (event: { keyCode: number }) => {
+      if (event.keyCode === 27) {
+        setQuery('');
+      }
+    };
+    window.addEventListener('keydown', handleEsc);
+    return () => {
+      window.removeEventListener('keydown', handleEsc);
+    };
+  }, []);
+
+  const fetchData = async (query: string) => {
+    try {
+      const response: { results: Row[] } = await appFetch(
+        `/api/rest/v3/disclosure-typeahead/?fullname=${query}&order_by=name_last`
+      );
+      const results: Row[] = response['results'];
+      setQuery(query);
+      setData(results);
+
+      console.log(response['results']);
```

Comment:
Let's yank this. It's a lot to log and you can catch it in the debugger if need be.